modules/transport-netty4/src/internalClusterTest/java/org/elasticsearch/http/netty4/Netty4ChunkedEncodingIT.java

```diff
@@ -79,22 +88,64 @@ public void testEmptyResponse() throws IOException {
     }
 
     private static void getAndCheckBodyContents(String route, String expectedBody) throws IOException {
-        final var response = getRestClient().performRequest(new Request("GET", route));
-        assertEquals(200, response.getStatusLine().getStatusCode());
-        assertThat(response.getEntity().getContentType().toString(), containsString(TEXT_CONTENT_TYPE));
-        if (Strings.hasLength(expectedBody)) {
-            assertTrue(response.getEntity().isChunked());
-        } // else we might have no chunks to send which doesn't need chunked-encoding
-        final String body;
-        try (var reader = new InputStreamReader(response.getEntity().getContent(), StandardCharsets.UTF_8)) {
-            body = Streams.copyToString(reader);
+        try (var ignored = withResourceTracker()) {
+            final var response = getRestClient().performRequest(new Request("GET", route));
+            assertEquals(200, response.getStatusLine().getStatusCode());
+            assertThat(response.getEntity().getContentType().toString(), containsString(TEXT_CONTENT_TYPE));
+            if (Strings.hasLength(expectedBody)) {
+                assertTrue(response.getEntity().isChunked());
+            } // else we might have no chunks to send which doesn't need chunked-encoding
+            final String body;
+            try (var reader = new InputStreamReader(response.getEntity().getContent(), StandardCharsets.UTF_8)) {
+                body = Streams.copyToString(reader);
+            }
+            assertEquals(expectedBody, body);
         }
-        assertEquals(expectedBody, body);
     }
 
+    public void testClientCancellation() {
+        try (var ignored = withResourceTracker()) {
+            final var cancellable = getRestClient().performRequestAsync(
+                new Request("GET", YieldsChunksPlugin.INFINITE_ROUTE),
+                new ResponseListener() {
+                    @Override
+                    public void onSuccess(Response response) {
+                        fail("should not complete");
+                    }
+
+                    @Override
+                    public void onFailure(Exception exception) {
+                        assertThat(exception, instanceOf(CancellationException.class));
+                    }
+                }
+            );
+            if (randomBoolean()) {
+                safeSleep(10);
```

Comment:
This seems to be too short. Locally it did not allow any chunked response to processed at all. Maybe we could do something like `safeSleep(randomLongBetween(10, 500));`?

Reply:
10ms is enough for me when running a few iterations, but yeah I can make this longer sometimes.